src/protocols/secure_channel/CASESession.cpp

```diff
@@ -228,8 +239,16 @@ class CASESession::WorkHelper
         // Hold strong ptr while work is outstanding
         helper->mStrongPtr.swap(strongPtr);
         auto status = DeviceLayer::PlatformMgr().ScheduleWork(AfterWorkHandler, reinterpret_cast<intptr_t>(helper));
```

Comment:
How about `VerifyOrDie()`? Device is anyway is bricked and need reboot, we do an intentional one. @bzbarsky-apple @mlepage-google @tcarmelveilleux

Reply:
Device is not "bricked"; it just can't have new CASE sessions established for it.

What do you think of my suggestion above of requiring at least one of locking the Matter stack and infallible dispatch to the Matter queue whenever this background crypto stuff is enabled?

Or even just requiring locks to work whenever background crypto is enabled?